app/models/thread.php

```diff
@@ -0,0 +1,88 @@
+<?php
+class Thread extends AppModel                    
+{
+    public $validation = array('title'=>array('length'=>array('validate_between',1,30),),); 
+
+    public static function getAll($offset, $limit)                
+    {
+        $threads = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM thread LIMIT {$offset}, {$limit}");
+        
+        foreach ($rows as $row) {                    
+            $threads[] = new Thread($row);
+        }
+        
+        return $threads;
+    }   
+
+    public function create(Comment $comment)
+    {
+        $this->validate();
+        $comment->validate();
+        if( $this->hasError() || $comment->hasError()){
+            throw new ValidationException('Invalid Thread or Comment');
+        }
+
+        $db = DB::conn();
+        $db->begin();
+        
+        $db->query('INSERT INTO thread SET title=?, created=NOW()',array($this->title));
+        $this->id = $db->lastInsertId();
+        $this->write($comment);
+
+        $db->commit();
+
+    }
+
+    public static function get($id)            
+    {
+        $db = DB::conn();
+
+        $row = $db->row('SELECT * FROM thread WHERE id = ?', array($id));
+
+        if(!$row) {
+            throw new RecordNotFoundException('no record found');
+        }
+
+        return new self($row);                    
+    }
+
+    public function getComments()
+    {
```

Comment:
I think this function must belong to Comment model? 
You are selecting data from comment table not in thread.